onnxscript/rewriter/generic_pattern.py

```diff
@@ -102,61 +82,13 @@ def _to_match_result(pmr: PatternMatchResult) -> orp.MatchResult:
     result = orp.MatchResult(success=True)
     result.nodes.extend(pmr.model_nodes)
     for var, val in pmr.matched_pattern_to_model_value.items():
-        result.bind(var, val)
-    result.outputs.extend(
-        [pmr.matched_pattern_to_model_value[v.name] for v in pmr.pattern_outputs]
-    )
+        if var.name is not None:
+            result.bind(var.name, val)
+    result.outputs.extend([pmr.matched_pattern_to_model_value[v] for v in pmr.pattern.outputs])
```

Comment:
What happens when var.name is None?

Reply:
Consider an intermediate statement in a pattern like below
```py
     XY = op.MatMul (X, Y) 
```
The user will not have access to this intermediate value `XY` in their condition-function for validating the match. But if they want access to it, they can indicate that by specifying an output-name for this as below:
```py
     XY = op.MatMul (X, Y, output=["XY"]) 
```
Then, they be able to access it in their condition function as below:
```py
   def check_condition(context, XY, ...):
```
In the first case, var.name is None in the pattern, and we will not bind any value in the match-binding we create. In the second case, we will.

Reply:
@shubhambhokare1 : this is worth documenting eventually, once the PR is merged.